src/isar/state_machine/states/send.py

```diff
@@ -26,70 +40,84 @@ def start(self):
 
     def stop(self):
         self.send_failure_counter = 0
+        if self.send_thread:
+            self.send_thread.join()
+        self.send_thread_output = SendThreadOutput()
+        self.send_thread = None
 
     def _run(self):
         while True:
             if self.state_machine.should_stop():
                 self.state_machine.stop_mission()
-            if not self.state_machine.status.mission_in_progress:
                 next_state: States = States.Cancel
                 break
 
             if not self.state_machine.status.mission_schedule.mission_steps:
                 next_state: States = States.Cancel
                 break
 
-            self.state_machine.status.current_mission_step = self._get_current_mission()
-            next_state: States = self._send_mission(
-                self.state_machine.status.current_mission_step
-            )
-            if not next_state == States.Send:
-                break
-
-            if self.state_machine.robot.mission_scheduled():
-                next_state: States = States.Monitor
-                break
+            if self.state_machine.should_send_status():
+                self.state_machine.send_status()
 
-            self.send_failure_counter += 1
-            self.logger.info("sending failed #: " + str(self.send_failure_counter))
-            if self.send_failure_counter >= self.send_failure_counter_limit:
-                self.logger.error(
-                    f"Failed to send mission after {self.send_failure_counter_limit} attempts. Cancelling mission."
+            if not self.send_thread:
+                self.state_machine.status.current_mission_step = (
+                    self._get_current_mission()
                 )
-                next_state: States = States.Cancel
-                break
+                self.send_thread = threading.Thread(
+                    target=send_mission_thread,
+                    args=(
```

Comment:
Could we list which input each of these would be, for instance: schedule_step_func=self.state_machine.robot.schedule_step?